src/CodeGen_RISCV.cpp

```diff
@@ -355,30 +364,24 @@ bool CodeGen_RISCV::call_riscv_vector_intrinsic(const RISCVIntrinsic &intrin, co
     llvm::Function *llvm_intrinsic =
         get_llvm_intrin(llvm_ret_type, mangled_name, llvm_arg_types);
 
+    // TODO: Should handle intrinsics other than binary operators.
+    // Call the LLVM intrinsic.
+    int actual_lanes = op->type.lanes();
+    llvm::Constant *actual_vlen = llvm::ConstantInt::get(xlen_type, actual_lanes);
+
     // Set vector fixed-point rounding flag if needed for intrinsic.
-    bool round_down = intrin.flags & RoundDown;
-    bool round_up = intrin.flags & RoundUp;
     if (round_down || round_up) {
```

Comment:
I'd probably move to building the vector of arguments and doing one call to the builder, but it's your call.